exporter/otlpexporter/consume_contract_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otlpexporter
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"go.opentelemetry.io/collector/config/confignet"
+
+	"github.com/cenkalti/backoff/v4"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+)
+
+const DefaultOTLPPort = 9999
```

Comment:
This one should not be exported as well

Reply:
btw why 9999? We can extract the real default port from `defaultGRPCEndpoint`

Reply:
I think it was a leftover form testing, now I left default grpc settings in the constructor, but to use `defaultGRPCEndpoint` in the config passed to exporter I think I'd need to make it exportable, right? I hardcoded it for now, I'm 99% sure it's wrong but don't know what would be clean way to do this.

Reply:
Or do you mean that I should use it like this?

Reply:
Keeping it hardcoded is ok